stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/factory/StreamOfflinePluginFactory.kt

```diff
@@ -76,6 +81,12 @@ public class StreamOfflinePluginFactory(
      */
     override fun get(user: User): Plugin = createOfflinePlugin(user)
 
+    private var repositoryFactory: RepositoryFactory? = null
+
+    public fun setRepositoryFactory(repositoryFactory: RepositoryFactory) {
+        this.repositoryFactory = repositoryFactory
+    }
+
```

Comment:
Could we potentially pass something like `RepositoryFactoryCreator` into `StreamOfflinePluginFactory` constructor with a default implementation provided? What do you think?

Reply:
We already default to `DatabaseRepositoryFactory`, if the user doesn't set a custom factory. That's the behavior that I expect to happen in most cases. 
We can ask that `RepositoryFactory` is passed in the constructor of `StreamOfflinePluginFactory`, but that would complicate this class, as users of Java can't use default values and they will be obligated to pass the `RepositoryFactory` inside `StreamOfflinePluginFactory`. In the current way, they would simply use the default factory.

Reply:
Just curious if we could use `StreamOfflinePluginFactory @JvmOverloads constructor(...)`.
What do you think? :)

Reply:
`@JvmOverloads` Would indeed solve this problem. Another problem that I see is that `DatabaseRepositoryFactory` need `User` which is only obtained when `StreamOfflinePluginFactory.get` is called, so it isn't available when the constructor is called

Reply:
Yup, that's why I thought we might have something like `RepositoryFactoryCreator` and pass it to `StreamOfflinePluginFactory` constructor with `@JvmOverloads`:
```kotlin
interface RepositoryFactoryCreator {
    fun createRepositoryFactory(
            scope: CoroutineScope,
            context: Context,
            user: User?,
            offlineEnabled: Boolean,
    ): RepositoryFactory
}
```

Just thinking out loud :)